data-processing/entities/sentences/extractor.py

```diff
@@ -45,22 +45,27 @@ def parse(self, tex_path: str, tex: str) -> Iterator[Sentence]:
 
         if last_segment is not None:
             plaintext_to_tex_offset_map[len(plaintext)] = last_segment.tex_end
-
         # Segment the plaintext. Return offsets for each setence relative to the TeX input
         segmenter = pysbd.Segmenter(language="en", clean=False, char_span=True)
+        # Record the current length of the plain text so account for the extractor bug
+        length_so_far_in_plain_text = 0
         for i, sentence in enumerate(segmenter.segment(plaintext)):
+            # Since the sentence extractor has several bugs related to finding start and end indicies
+            # we will simply set them ourselves.
+            real_start = plaintext.find(sentence.sent, length_so_far_in_plain_text)
+            start = plaintext_to_tex_offset_map[real_start]
+            real_end = real_start + (sentence.end - sentence.start)
+            end = plaintext_to_tex_offset_map[real_end]
+            length_so_far_in_plain_text = real_end
 
-            start = plaintext_to_tex_offset_map[sentence.start]
-            end = plaintext_to_tex_offset_map[sentence.end]
-            tex = tex[start:end]
+            tex_sub = tex[start:end]
```

Comment:
also I was pretty sure this was a bug so I went ahead and changed it, but the tex/tex_context were blank for everything that wasn't the first string, was this intentional?

Reply:
Oh yikes! Yes that was definitely a bug 🐞 . Great catch!